.github/workflows/krel-release-notes-validate.yaml

```diff
@@ -0,0 +1,140 @@
+          cd "$TEMP_DIR"
+          
+          echo "Downloading latest krel version $KREL_VERSION..."
+          if ! curl -sL "https://github.com/kubernetes/release/releases/download/$KREL_VERSION/$ARTIFACT_NAME" -o krel; then
+              echo "Failed to download krel"
+              exit 1
+          fi
+          
+          KREL_CERT="https://github.com/kubernetes/release/releases/download/$KREL_VERSION/$ARTIFACT_NAME.pem"
+          KREL_SIG="https://github.com/kubernetes/release/releases/download/$KREL_VERSION/$ARTIFACT_NAME.sig"
+          
+          echo "Using cosign to verify signature of krel version $KREL_VERSION"
+          if ! cosign verify-blob --certificate "$KREL_CERT" --signature "$KREL_SIG" \
+              --certificate-identity "https://github.com/kubernetes/release/.github/workflows/release.yml@refs/tags/$KREL_VERSION" \
+              --certificate-oidc-issuer "https://token.actions.githubusercontent.com" krel; then
+              echo "Signature verification failed for krel version: '$KREL_VERSION'"
+              exit 1
+          fi
+          
+          chmod +x krel
+          mkdir -p "$HOME/.local/bin"
+          mv krel "$HOME/.local/bin/"
+          cd - > /dev/null
+          rm -rf "$TEMP_DIR"
+          
+          KREL_PATH="$HOME/.local/bin/krel"
+          echo "krel-path=$KREL_PATH" >> "$GITHUB_OUTPUT"
+          echo "Krel installed at: $KREL_PATH"
+      - name: Run if releases YAML changes exist and validate the YAML
+        id: validate_releases_yaml
+        env:
+          KREL_PATH: ${{ steps.install-krel.outputs.krel-path }}
+        run: |
+          # Get a list of changed YAML files based on git diff
+          CHANGED_FILES=$(git diff --name-only ${{ github.event.pull_request.base.sha }} ${{ github.sha }} -- releases/ | grep -E '\.ya?ml$' || true)
+
+          if [ -n "$CHANGED_FILES" ]; then
+            echo "validating base sha ${{ github.event.pull_request.base.sha }}"
+
+            # Initialize an array for invalid files
+            INVALID_FILES=""
+
+            while IFS= read -r file; do
+              set +e
+              "${KREL_PATH}" release-notes validate --path-to-release-notes "$file"
+              exit_code=$?
+              set -e
+              
+              if [ $exit_code -ne 0 ]; then
+                  INVALID_FILES="${INVALID_FILES}- ${file##*/}\n"
+              fi
+            done <<< "$CHANGED_FILES"
+          
+            # If there are any invalid files, set the output and fail
+            if [ -n "$INVALID_FILES" ]; then
+              echo "invalid_files<<EOF" >> $GITHUB_OUTPUT
+              echo -e "$INVALID_FILES" >> $GITHUB_OUTPUT
+              echo "EOF" >> $GITHUB_OUTPUT
+              exit 1
+            else
+              echo "All YAML files are valid."
+            fi
+          else
+            echo "No YAML files changed under /releases/*"
+          fi
+      - name: Comment on PR if invalid yaml detected
```

Comment:
i think this will not work as expected because in fork PRs we dont have the `write` permission for the github token

i would add this as job summary and then we can see this in the checks tab

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#adding-a-job-summary

Reply:
I agree. It only needs `pull-requests` but I'd prefer to keep the token with less permissions. Just fail it and we can check the results on the workflow run.

Reply:
Migrated to github job summary. Test run: https://github.com/npolshakova/sig-release/actions/runs/12806142763?pr=4